linera-execution/src/runtime.rs

```diff
@@ -617,54 +592,23 @@ impl<const W: bool> BaseRuntime for SyncRuntimeInternal<W> {
         receiver.recv_response()
     }
 
-    // TODO(#1152): simplify away
-    fn lock_new(&mut self) -> Result<Self::Lock, ExecutionError> {
-        let id = self.application_id()?;
-        let state = self.view_user_states.entry(id).or_default();
-        ensure!(!state.locked, ExecutionError::ApplicationIsInUse(id));
-        state.locked = true;
-        Ok(())
-    }
-
-    fn lock_wait(&mut self, _promise: &Self::Lock) -> Result<(), ExecutionError> {
-        Ok(())
-    }
-
-    fn unlock_new(&mut self) -> Result<Self::Unlock, ExecutionError> {
-        let id = self.application_id()?;
-        let state = self.view_user_states.entry(id).or_default();
-        ensure!(state.locked, ExecutionError::ApplicationStateNotLocked(id));
```

Comment:
Have you looked at removing the now-unused error variants?

Reply:
Oops! Fixed :sweat_smile: Thanks!